src/features/vote/ActiveViewDetailsModal.tsx

```diff
@@ -38,24 +43,85 @@ interface Props {
   timestamp: string;
   ancData: string;
   unix: string;
+  committedVotes: VoteEvent[] | undefined | void;
+  revealedEvents: VoteRevealed[] | undefined | void;
+  roundId: string;
 }
 
 const NULL_ANC_DATA = "0x";
+const NULL_NUM_COMMITTED_VOTES = 0;
+const NULL_PERCENTAGE = "0.00";
 
 const _ActiveViewDetailsModal: ForwardRefRenderFunction<
   HTMLElement,
   PropsWithChildren<Props>
 > = (
-  { isOpen, close, proposal, setModalState, timestamp, ancData, unix },
+  {
+    isOpen,
+    close,
+    proposal,
+    setModalState,
+    timestamp,
+    ancData,
+    unix,
+    committedVotes,
+    revealedEvents,
+    roundId,
+  },
   externalRef
 ) => {
+  const [numberCommittedVotes, setNumberCommittedVotes] = useState(
+    NULL_NUM_COMMITTED_VOTES
+  );
+  const [numberRevealedAddresses, setNumberRevealedAddresses] = useState(0);
+  const [revealPercentage, setRevealPercentage] = useState(NULL_PERCENTAGE);
   const [copySuccess, setCopySuccess] = useState(false);
   // Note: because there is dynamic content, this will rebuild the tooltip for addressing the conditional
   // elements on the page. See ReactTooltip docs.
   useEffect(() => {
     ReactTooltip.rebuild();
   });
 
+  useEffect(() => {
+    if (committedVotes && committedVotes.length && roundId && proposal) {
+      const findCommitsForProposal = committedVotes.filter(
+        (x) =>
+          x.identifier === proposal && x.roundId === roundId && x.time === unix
+      );
+      const uniqueCommitsByAddress = findCommitsForProposal.filter(
+        (v, i, a) => a.findIndex((t) => t.address === v.address) === i
+      );
+      setNumberCommittedVotes(uniqueCommitsByAddress.length);
+    } else {
+      setNumberCommittedVotes(NULL_NUM_COMMITTED_VOTES);
+    }
+  }, [committedVotes, roundId, proposal, unix]);
+
+  useEffect(() => {
+    if (revealedEvents && revealedEvents.length && roundId && proposal) {
+      const findRevealsForProposal = revealedEvents.filter(
+        (x) =>
+          x.identifier === proposal && x.roundId === roundId && x.time === unix
+      );
+
+      setNumberRevealedAddresses(findRevealsForProposal.length);
+    } else {
+      setNumberRevealedAddresses(0);
+    }
+  }, [revealedEvents, roundId, proposal, unix]);
+
+  useEffect(() => {
+    if (numberCommittedVotes > 0) {
+      const percentage = (
+        (numberRevealedAddresses / numberCommittedVotes) *
+        100
+      ).toFixed(2);
+      setRevealPercentage(percentage);
```

Comment:
these stats like functions seem appropriate for a utility file that can be unit tested, though they might not be complex enough to warrant that. I could imagine a collection of static functions which can then be composed into a function that returns the end state you want, ie the final display variables. wdyt?

Reply:
Yup, this could be abstracted out into a utility function. I'm not sure it would be used elsewhere, so it's more up to convention if we want the code close (IE: in this modal) or pulled out for testing.